make/autoconf/lib-bundled.m4

```diff
@@ -216,7 +221,7 @@ AC_DEFUN_ONCE([LIB_SETUP_ZLIB],
   LIBZ_CFLAGS=""
   LIBZ_LIBS=""
   if test "x$USE_EXTERNAL_LIBZ" = "xfalse"; then
-    LIBZ_CFLAGS="$LIBZ_CFLAGS -I$TOPDIR/src/java.base/share/native/libzip/zlib"
+    LIBZ_CFLAGS="$LIBZ_CFLAGS $APPLE_LIBZ_CFLAGS -I$TOPDIR/src/java.base/share/native/libzip/zlib"
```

Comment:
Declaring APPLE_LIBZ_CFLAGS far away (and only conditionally) and then using it once here just makes for hard-to-read code. 

```suggestion
    LIBZ_CFLAGS="$LIBZ_CFLAGS -I$TOPDIR/src/java.base/share/native/libzip/zlib"
    if test "x$OPENJDK_TARGET_OS" = xmacosx; then
      LIBZ_CFLAGS="$LIBZ_CFLAGS -DHAVE_UNISTD_H"
    fi
```
... and remove the assignment above.

Reply:
Updated the PR to implement this suggestion